server/game/core/attack/AttackFlow.ts

```diff
@@ -45,6 +45,11 @@ export class AttackFlow extends BaseStepWithPipeline {
 
     private declareAttack() {
         this.attack.attacker.registerAttackKeywords();
+        this.attack.attacker.setActiveAttack(this.attack);
+        if (this.attack.target.isUnit()) {
+            // TODO: should we move this to a place where it can be set on bases?
```

Comment:
Oh yeah that will actually be relevant really soon for stuff like Ephant Mon... plus it will cause the behavior to be inconsistent if people write other code expecting this to work correctly for bases. Thankfully it's an easy fix, we can just move the `_activeAttack` field and the `defending` property into the `WithDamage` mixin (since everything that has the damage property could possibly participate in an attack and be a target) and then we leave the `attacking` property on `UnitProperties` since that's the only kind of thing that can be an attacker.

That should be a simple copy-paste that just works out of the box.